kong/plugins/jwt/access.lua

```diff
@@ -0,0 +1,112 @@
+local jwt = require "luajwt"
+local cache = require "kong.tools.database_cache"
+local responses = require "kong.tools.responses"
+local constants = require "kong.constants"
+local string_format = string.format
+local select = select
+local dao = dao
+local re_gmatch = ngx.re.gmatch
+
+local _M = {}
+
+--- Retrieve a JWT in a request.
+-- Checks for the JWT in URI parameters, then in the `Authorization` header.
+-- @param request ngx request object
+-- @param conf Plugin configuration
+-- @return token JWT token contained in request or nil
+-- @return err
+local function retrieve_token(request, conf)
+  local uri_parameters = request.get_uri_args()
+
+  for _, v in ipairs(conf.uri_param_names) do
+    if uri_parameters[v] then
+      return uri_parameters[v]
+    end
+  end
+
+  local authorization_header = request.get_headers()["authorization"]
+  if authorization_header then
+    local iterator, iter_err = re_gmatch(authorization_header, "\\s*[Bb]earer\\s*(.+)")
```

Comment:
Shouldn't the whitespace between Bearer and the token be required?  You don't want to match `BearerToken` or anything weird. `\\s+` instead of `\\s*`

I also question the whitespace at the start of the header, but I'll digress on that one.

I would recommend `"\\s*[Bb]earer\\s+(.+)"